signer/src/testing/transaction_coordinator.rs

```diff
@@ -526,6 +540,147 @@ where
         assert_eq!(signer_utxo, expected);
     }
 
+    /// Assert we get the correct UTXO in case of donations
+    pub async fn assert_get_signer_utxo_donations(mut self) {
+        let mut rng = rand::rngs::StdRng::seed_from_u64(46);
+        let network = network::in_memory::Network::new();
+        let signer_info = testing::wsts::generate_signer_info(&mut rng, self.num_signers);
+
+        let mut signer_set =
+            testing::wsts::SignerSet::new(&signer_info, self.signing_threshold as u32, || {
+                network.connect()
+            });
+
+        let (aggregate_key, bitcoin_chain_tip, mut test_data) = self
+            .prepare_database_and_run_dkg(&mut rng, &mut signer_set)
+            .await;
+
+        let original_test_data = test_data.clone();
+
+        // The scenario is:
+        // [bitcoin_chain_tip] +- [block a1 with signer utxo] - [block a2 with donation]
+        //                     +- [block b1 with donation]
```

Comment:
In this diagram, does block a2 have greater height than block a1?

Reply:
Yes -- changed the `bitcoin_chain_tip` to make it clear it refers to the "original" chain tip before adding the test blocks